terraform/main.tf

```diff
@@ -0,0 +1,499 @@
+# Global Configuration
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.24.0"
+    }
+    archive = {
+      source  = "hashicorp/archive"
+      version = "~> 2.2.0"
+    }
+  }
+
+  required_version = "~> 1.0"
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+
+# Account that has access to the organization root to get the account list
+provider "aws" {
+  alias      = "management"
+  region     = var.aws_region
+  access_key = var.AWS_MANAG_ACCESS_KEY_ID
```

Comment:
This authentication method works when deploying locally with the ephemeral credentials but won't work from CI/CD. We can leave this as is for now but we will need to re-factor once we setup CI/CD to use role assumption instead.

Reply:
I've looked a bit to implement and comment the assume role but I don't have at most I can but a placeholder right now so it's might not be worth it ?

Reply:
Yeah this was more just a comment for where we will go in the future. We can leave this as is for now.